File: library/src/main/java/com/yarolegovich/slidingrootnav/SlidingRootNav.java
package com.yarolegovich.slidingrootnav;

import com.yarolegovich.slidingrootnav.util.BadgeDrawerToggle;

/**
 * Created by yarolegovich on 25.03.2017.
 */

public interface SlidingRootNav {

    boolean isMenuClosed();

    boolean isMenuOpened();

    boolean isMenuLocked();

    void closeMenu();

    void closeMenu(boolean animated);

    void openMenu();

    void openMenu(boolean animated);

    void setMenuLocked(boolean locked);

    SlidingRootNavLayout getLayout();

    BadgeDrawerToggle getBadgeDrawerToggle();
}

File: library/src/main/java/com/yarolegovich/slidingrootnav/SlidingRootNav.kt
package com.yarolegovich.slidingrootnav

/**
 * Created by yarolegovich on 25.03.2017.
 */

public interface SlidingRootNav {

    val isMenuClosed: Boolean

    val isMenuOpened: Boolean

    var isMenuLocked: Boolean

    val layout: SlidingRootNavLayout

    fun closeMenu()

    fun closeMenu(animated: Boolean)

    fun openMenu()

    fun openMenu(animated: Boolean)

}
